### PR TITLE
Add an example using Response body

### PR DESCRIPTION
**Pull request by yutakahirano**
Addressing #118.

### CODE REVIEW

**Comment by annevk** [comment]
Can you also update the initial paragraph of https://fetch.spec.whatwg.org/#fetch-api to say "lacking when it comes to request progression (not response progression)"?

**Comment by annevk** [comment]
@domenic could you have a look at the example just in case?

**Comment by domenic** [comment]
I really preferred the example before https://github.com/yutakahirano/fetch-with-streams/pull/58. It was more idiomatic promise usage and much clearer. The version in this PR is an example of the [explicit promise construction anti-pattern](http://stackoverflow.com/q/23803743/3191).

But given that naive implementation of the promise spec currently causes the more idiomatic example to retain some memory for each chunk, maybe the proposed example is better since it avoids those issues in current implementations. I am a bit torn. Since @yutakahirano thinks this proposed version is better than I am OK with it.

File: Overview.src.html [hunk 1]
```diff
@@ -3082,6 +3082,31 @@ <h4 id="should-response-to-request-be-blocked-due-to-nosniff?"><dfn title="shoul
...
+  var total = 0
+  return new Promise((resolve, reject) => {
+    function pump() {
+      reader.read().then(({done, value}) => {
+        if (done) {
+          resolve()
+          return
+        }
+        total += value.byteLength
+        log("received " + value.byteLength + " bytes (" + total + " bytes in total).")
```
Comment (domenic) [inline Overview.src.html]: Optional ES6 goodness: ``log(`received ${value.byteLength} bytes (${total} bytes in total)`)``.

Comment (yutakahirano) [inline Overview.src.html]: Thanks, done.

**Comment by yutakahirano** [comment]
> Can you also update the initial paragraph of https://fetch.spec.whatwg.org/#fetch-api to say "lacking when it comes to request progression (not response progression)"?

Done.

Regarding the example, code, I agree that the previous one is more elegant and it's unfortunate that existing implementations cannot execute it efficiently. That said, we received a report related with this problem once, so encouraging the pattern is dangerous at least for now, I think.

**Comment by annevk** [comment]
Rebase? LGTM.

**Comment by yutakahirano** [comment]
Rebased.

**Pull request merged**